src/UI/templates/js/MainControls/mainbar.js

```diff
@@ -107,23 +107,42 @@ il.UI.maincontrols = il.UI.maincontrols || {};
 			},
 			init = function(inititally_active) {
 				var mb = il.UI.maincontrols.mainbar,
-					cookie_state = mb.persistence.read();
+					cookie_state = mb.persistence.read(),
+					init_state = mb.model.getState();
 
+				//apply cookie-state
 				if(Object.keys(cookie_state).length > 0) {
-					mb.model.setState(cookie_state);
+					if(Object.keys(init_state.tools).length === 0) {
+						cookie_state.tools = {};
+					} else {
+						cookie_state.tools = init_state.tools;
+					}
```

Comment:
We could reduce that to `cookie_state.tools = init_state.tools;` because in the first case `init_state.tools` will equal `{}` anyway, right? Or am I missing something?

Reply:
this is true ;)